gemini_report_writer/agents/quality_pipeline.py

```
@@ -0,0 +1,582 @@
+        print(f"🧩 Coherence Quality: {overall_score:.2f} ({'✅ PASS' if stage_passed else '❌ FAIL'})")
+        return report
+    
+    def finalize_quality_report(self) -> SystemQualityReport:
+        """Finalize and return the complete quality report"""
+        if not self.current_report:
+            raise ValueError("No quality tracking session active")
+            
+        self.current_report.end_time = time.time()
+        
+        # Print final summary
+        print(f"\n📊 FINAL QUALITY REPORT")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)